### PR TITLE
feat(memory)!: support binary files as Uint8Array

### LINKED ISSUE #169: Support binary files
`IDirectoryContents` interface used by `createMemoryFs` and `populateDirectory` only supports strings for file contents, which means it can't be used for binary files¹.

To represent binary files we could use an `Uint8Array`, which is already compatible with `Buffer` when using the native `fs` module:

```ts
Buffer.from([]) instanceof Uint8Array; // => true

const data: Uint8Array = fs.readFileSync('foo.txt');
fs.writeFileSync('bar.txt', data);
fs.writeFileSync('bar.txt', new Uint8Array([65, 66, 67]));
```

--- 
[1] Strictly speaking, JS strings can represent binary data without overhead, since they're sequences of 16-bit code units.

```js
String.fromCharCode(0).charCodeAt(0) // => 0
String.fromCharCode(65535).charCodeAt(0) // => 65535
String.fromCharCode(65536).charCodeAt(0) // => 0, wrap around
String.fromCharCode(-1).charCodeAt(0) // 65535, wrap around
```

But this would require converting to and from unit8 arrays when working with `readFile` / `writeFile`, and we wouldn't be able to tell based on the string alone if conversion is required. So this sucks. And also files with an odd number of bytes can't be represented in this way.

### PR DESCRIPTION
**Pull request by AviVahl**
using `TextEncoder`/`TextDecoder` for conversion.

closes #169

### CODE REVIEW

**Comment by barak007** [comment]
This PR will make this library "slow as hell".
I think this is the wrong choice. Cache must be implemented otherwise I don't feel this is usable. 
I cannot think about why not merging the cache version of the implementation what was wrong with that?

File: packages/memory/src/types.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export type IFsMemNodeType = IFsMemFileNode | IFsMemDirectoryNode | IFsMemSymlin
 
 export interface IFsMemFileNode extends IFsMemNode {
   type: "file";
-  contents: string;
+  contents: Uint8Array;
```
Comment (alisey) [inline packages/memory/src/types.ts:33]: How about storing `string | Uint8Array` and then converting only if necessary in readFile / readFileSync?

Comment (barak007) [reply packages/memory/src/types.ts:33]: I did it but I'm not sure why it's not merged. #804

**Comment by AviVahl** [comment]
> This PR will make this library "slow as hell". I think this is the wrong choice. Cache must be implemented otherwise I don't feel this is usable. I cannot think about why not merging the cache version of the implementation what was wrong with that?

It was hard to review. Cannot tell your changes from the rest. It also had conflicts with the changes that landed main branch.

**Comment by AviVahl** [comment]
See https://github.com/wixplosives/file-services/pull/1141
Took ~5 mins now that this is in.

**Pull request merged**